jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/execution/sam/SamInitProjectWizardTest.kt

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.execution.sam
+
+import com.intellij.ide.projectWizard.ChooseTemplateStep
+import com.intellij.ide.projectWizard.NewProjectWizardTestCase
+import com.intellij.ide.projectWizard.ProjectSettingsStep
+import com.intellij.ide.projectWizard.ProjectTypeStep
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.projectRoots.JavaSdk
+import com.intellij.openapi.projectRoots.ProjectJdkTable
+import com.intellij.openapi.projectRoots.SdkType
+import com.intellij.testFramework.IdeaTestUtil
+import com.jetbrains.python.sdk.PythonSdkType
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.settings.SamSettings
+import software.aws.toolkits.jetbrains.ui.wizard.SamInitModuleBuilder
+import software.aws.toolkits.jetbrains.ui.wizard.SamInitTemplateSelectionStep
+import software.aws.toolkits.jetbrains.utils.rules.PyTestSdk3x
+import kotlin.reflect.KClass
+
+// JUnit3-style to take advantage of IDEA's test base
+class SamInitProjectWizardTest : NewProjectWizardTestCase() {
+    fun testExceptionIfSamNotConfigured() {
+        try {
+            createProject { step ->
+                if (step is ProjectTypeStep) {
+                    assertTrue(step.setSelectedTemplate("SAM", null))
+                    val builder = myWizard.projectBuilder as SamInitModuleBuilder
+                    builder.runtimeSelectionPanel.samExecutableField.text = ""
+                }
+            }
+            fail("Exception was not thrown")
+        } catch (e: RuntimeException) {
+            // expected a runtime exception
+            assertTrue(e.message == "SAM CLI executable not configured" ||
+                e.message == "com.intellij.ide.projectWizard.ProjectTypeStep is not validated")
+        }
+    }
+
+    fun testPython36Runtime() {
+        helloWorldTest(Runtime.PYTHON3_6, PythonSdkType::class)
+    }
+
+    fun testPython27Runtime() {
+        helloWorldTest(Runtime.PYTHON2_7, PythonSdkType::class)
+    }
+
+    fun testJavaRuntime() {
+        helloWorldTest(Runtime.JAVA8, JavaSdk::class)
+    }
+
+    fun helloWorldTest(runtime: Runtime, sdkType: KClass<out SdkType>) {
+        SamSettings.getInstance().savedExecutablePath = "sam"
+
+        createProject { step ->
+            val stepNum = myWizard.currentStep
+            when (step) {
```

Comment:
I'm a bit confused how this logic tests stuff...

Reply:
The `helloWorldTest` could be `private`

Reply:
This test should assert that we reach the reach the end of the create project flow and that changing the runtime switches the SdkSettingsStep to the appropriate SDK language